src/msgraph_core/requests/batch_request_item.py

```diff
@@ -238,16 +240,30 @@ def serialize(self, writer: SerializationWriter) -> None:
         Args:
             writer (SerializationWriter): The writer to write to.
         """
+        if not writer:
+            raise ValueError("writer cannot be None")
+
         writer.write_str_value('id', self.id)
         writer.write_str_value('method', self.method)
         writer.write_str_value('url', self.url)
+
         writer.write_collection_of_primitive_values('depends_on', self._depends_on)
-        headers = {key: ", ".join(val) for key, val in self._headers.items()}
+
+        headers = self._headers
         writer.write_collection_of_object_values('headers', headers)
+
         if self._body:
-            json_object = json.loads(self._body)
-            is_json_string = json_object and isinstance(json_object, dict)
-            writer.write_collection_of_object_values(
-                'body',
-                json_object if is_json_string else base64.b64encode(self._body).decode('utf-8')
-            )
+            if isinstance(self._body, bytes):
```

Comment:
Any chance we can capture in a test the example of a body parameter when its a stream and when its a json object?

Reply:
Updated